crates/re_types/definitions/rerun/archetypes/mesh3d.fbs

```diff
@@ -0,0 +1,81 @@
+include "fbs/attributes.fbs";
+
+include "rerun/datatypes.fbs";
+include "rerun/components.fbs";
+
+namespace rerun.archetypes;
+
+// ---
+
+/// A 3D triangle mesh as specified by its per-mesh and per-vertex properties.
+///
+/// \py Example
+/// \py -------
+/// \py
+/// \py Simple:
+/// \py ```python
+/// \py \include:../../../../../docs/code-examples/mesh3d_indexed.py
+/// \py ```
+/// \py
+/// \py Partial updates:
+/// \py ```python
+/// \py \include:../../../../../docs/code-examples/mesh3d_partial_updates.py
+/// \py ```
+///
+/// \rs ## Example
+/// \rs
+/// \rs Simple:
+/// \rs ```ignore
+/// \rs \include:../../../../../docs/code-examples/mesh3d_indexed.rs
+/// \rs ```
+/// \rs
+/// \rs Partial updates:
+/// \rs ```ignore
+/// \rs \include:../../../../../docs/code-examples/mesh3d_partial_updates.rs
+/// \rs ```
+///
+/// \cpp ## Example
+/// \cpp
+/// \cpp Simple:
+/// \cpp ```
+/// \cpp \include:../../../../../docs/code-examples/mesh3d_indexed.cpp
+/// \cpp ```
+table Mesh3D (
+  "attr.rust.derive": "PartialEq",
+  order: 100
+) {
+  // --- Required ---
+
+  /// The positions of each vertex.
+  ///
+  /// If no `indices` are specified, then each triplet of positions is interpreted as a triangle.
+  vertex_positions: [rerun.components.Position3D] ("attr.rerun.component_required", order: 1000);
+
+  // --- Recommended ---
+
+  /// Optional properties for the mesh as a whole (including indexed drawing).
+  //
+  // NOTE: We cannot have triangle indices here as that would break our instance key rules (either 0, 1 or N).
+  mesh_properties: rerun.components.MeshProperties ("attr.rerun.component_recommended", nullable, order: 2000);
```

Comment:
Why not call this `mesh_indices: rerun.components.MeshIndices` instead?

Reply:
Feels likely like we're gonna want to stash a few more things in there sooner than later (winding order, index format, ...) :man_shrugging:

Reply:
`MeshTriangles` maybe? "properties" is very vague. I would expect there to be color stuff in there, not fundamental stuff like indices.

Reply:
Im not necessarily a fan of `MeshProperties` but I like `MeshTriangles` even less tbh.

- This really has nothing to do with triangles at this point (it's now `vertex_indices`)
- Indices are not "fundamental", quite the contrary they're optional :/

Anyway, time to ship this; feel free to change it in another PR.